support_branding/models/res_company.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Sunflower IT
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    @api.model
+    def get_ir_config_param_data(self, key):
```

Comment:
Anyone who can read the company record, can now call this function, so it's not very safe - `ir.config.parameter` can contain all kinds of stuff: API keys, passwords, ... we don't want all portal users to be able to read those.

What I meant with "add these settings to a proper res.config.settings screen and then get them by reading from the company record instead", was, that we define the `support_company_*` fields on `res.company`. But that was just one idea out of many possibilities. Maybe it's enough if you just define `get_support_branding_data` instead, which gives out a dictionary with _only_ the support branding keys, of which it is OK if everyone reads them. Or maybe just backend users is enough:

```
if not user.has_group('base.group_user'):
    raise AccessError(...)
```

Reply:
@thomaspaulb added the above and a unit test for it testing with portal user demo

Reply:
Still not secure. Now, anyone who's an internal user, can read all key/value pairs in ir.config.parameter. You need to make it into a function that returns only the support_branding keys.

Reply:
@thomaspaulb I switched to key/value pair based on search for support_branding values as they are stored using the prefix `support_` so query to `key ilike %support_%` returns the values then I use them later.